modules/ipromBidAdapter.js

```diff
@@ -0,0 +1,72 @@
+import * as utils from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+
+const BIDDER_CODE = 'iprom';
+const ENDPOINT_URL = 'https://core.iprom.net/programmatic';
+const VERSION = 'v1.0.0';
+const DEFAULT_CURRENCY = 'EUR';
+const DEFAULT_NETREVENUE = true;
+const DEFAULT_TTL = 360;
+
+export const spec = {
+  code: BIDDER_CODE,
+  isBidRequestValid: function ({ bidder, params = {} } = {}) {
+    // id parameter checks
+    if (!params.id) {
+      utils.logError(`${bidder}: Parameter 'id' missing`);
+      return false;
+    } else if (typeof params.id !== 'string') {
+      utils.logError(`${bidder}: Parameter 'id' needs to be a string`);
+      return false;
+    }
+    // dimension parameter checks
+    if (!params.dimension) {
+      utils.logError(`${bidder}: Required parameter 'dimension' missing`);
+      return false;
+    } else if (typeof params.dimension !== 'string') {
+      utils.logError(`${bidder}: Parameter 'dimension' needs to be a string`);
+      return false;
+    }
```

Comment:
regex check for `wxh` or is that too much?

Reply:
Dimension parameter is awkwardly named and does not just include `wxh` dimensions but some other strings as well.

We plan to fully migrate to using Sizes but dimension is here for backwards compatibility with our ad server for now.